### PR TITLE
docs: Add links checker

### PR DESCRIPTION
**Pull request by Fokko**

### CODE REVIEW

File: format/spec.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ In addition to row-level deletes, version 2 makes some requirements stricter for
 
 ## Overview
 
+<!-- markdown-link-check-disable-next-line -->
```
Comment (nastra) [inline format/spec.md:60]: should this link maybe have `https://iceberg.apache.org/` as its prefix as it's already being used in other places for images?

File: .github/workflows/docs-check-links.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Check Markdown docs links
+
+on:
+  push:
+    paths:
+      - docs/**
```
Comment (manuzhang) [inline .github/workflows/docs-check-links.yml:25]: `site/docs/**` as well?

Comment (Fokko) [reply .github/workflows/docs-check-links.yml:25]: Good one, thanks!

**Review by nastra — APPROVED** [review]
LGTM, this is really great to have, thanks @Fokko

**Review by manuzhang — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks for the review @nastra and @manuzhang

**Comment by manuzhang** [comment]
Will this fix get deployed to docs site immediately?

**Comment by print-sid8** [comment]
Hi guys,

Just reporting something I have noticed in the docs website of Iceberg for versions 1.5.0 and 1.5.1 and also the 'latest' section

Lets say I go to - 
https://iceberg.apache.org/docs/1.5.0/
and I click on the hyperlink that is in the following sentence , which points to another section of the documentation -

```
Iceberg will convert the column type in Spark to corresponding Iceberg type. Please check the section of [type compatibility on creating table](https://iceberg.apache.org/docs/1.5.0/spark-getting-started.md#spark-type-to-iceberg-type) for details.
```

I guess its visible in Markdown that most such "internal hyperlink" ie., links that points to other sections of Iceberg documentation have ".md" in the URLs , such as -

[ https://iceberg.apache.org/docs/1.5.0/spark-getting-started.md#spark-type-to-iceberg-type ,
https://iceberg.apache.org/docs/1.5.0/configuration.md , 
https://iceberg.apache.org/docs/1.5.0/partitioning.md ] and more.

all the above links with ".md" give 404 error - The requested URL was not found on this server.

and so i removed the ".md" and they all go to right sections/webpages.

[ https://iceberg.apache.org/docs/1.5.0/spark-getting-started#spark-type-to-iceberg-type ,
https://iceberg.apache.org/docs/1.5.0/configuration/ , 
https://iceberg.apache.org/docs/1.5.0/partitioning/ ] 


This is seen in all the docs pages for 1.5.0 , 1.5.1 and 'latest'  , but not in "nightly" pages

example -
[https://iceberg.apache.org/docs/1.5.1/configuration.md, https://iceberg.apache.org/docs/latest/configuration.md ]

**Comment by nastra** [comment]
@bitsondatadev could you take a look at the issue that @print-sid8 reported please?

**Comment by manuzhang** [comment]
@nastra and @print-sid8 These are known issues for previous versions. I'm a bit surprised it's not fixed in 1.5.1 which was just released.

**Comment by manuzhang** [comment]
I back-ported the fix to 1.5.x releases in #10411

**Pull request merged**